src/app/events/schemas/FrontMatterSchema.ts

```diff
@@ -9,15 +9,24 @@ import { ScheduleSchema } from './ScheduleSchema'
 import { SpeakersSchema } from './SpeakerSchema'
 import { SponsorsSchema } from './SponsorSchema'
 
-const { validCategoryIds } = getEventsCategorySettings()
+const { validIds: validCategoryIds } = getFieldSettings({
+  collectionName: 'event_entries',
+  fieldName: 'category',
+})
+const { validIds: validRegionIds } = getFieldSettings({
+  collectionName: 'event_entries',
+  fieldName: 'region',
+})
 
 const CategorySchema = createEnumSchema(validCategoryIds)
+const RegionSchema = createEnumSchema(validRegionIds)
 
 export const EventFrontMatterSchema = DynamicBaseDataSchema.extend({
   title: z.string(),
   category: CategorySchema,
   description: z.string().optional(),
   location: z.string(),
+  region: RegionSchema.optional(),
```

Comment:
@CharlyMartin - What do you think about this? This is now `optional` because an event can be virtual.

Reply:
I think we can enforce `region` to be present if location is not `"Virtual"`, let me have a look.

Reply:
What about something like this?

```ts
const BaseEventFrontMatterSchema = DynamicBaseDataSchema.extend({
  title: z.string(),
  category: CategorySchema,
  description: z.string().optional(),
  'external-link': z
    .object({
      url: z.string().url(),
      text: z.string().optional(),
    })
    .optional(),
  'luma-calendar-link': z.string().url().optional(),
  'luma-events-section': z
    .object({
      title: z.string().optional(),
      'embed-link': z.string().url(),
    })
    .optional(),
  'start-date': z.coerce.date(),
  'end-date': z.coerce.date().optional(),
  schedule: ScheduleSchema.optional(),
  speakers: SpeakersSchema.optional(),
  sponsors: SponsorsSchema.optional(),
  'recap-youtube-embed-url': z.string().url().optional(),
  'recap-youtube-playlist-url': z.string().url().optional(),
}).strict()

const VirtualEventSchema = BaseEventFrontMatterSchema.extend({
  location: z.literal('Virtual'),
  region: z.void(),
}).strict()

const InPersonEventSchema = BaseEventFrontMatterSchema.extend({
  location: z.string(),
  region: RegionSchema,
}).strict()

export const EventFrontMatterSchema = z.union([
  VirtualEventSchema,
  InPersonEventSchema,
])
```

Reply:
It doesn't throw an error if the `primary` location is "Virtual" and the user fills out the `region`... Maybe we try with [refine](https://zod.dev/?id=refine)?

Reply:
```ts
const LocationSchema = z
  .object({
    primary: z.string(),
    region: RegionSchema.optional(),
  })
  .refine((data) => !(data.primary === 'Virtual' && data.region), {
    message: 'Region must be empty if the event is virtual.',
    path: ['region'],
  })
  .refine((data) => !(data.primary !== 'Virtual' && !data.region), {
    message: 'Region must be filled out if the event is not virtual.',
    path: ['region'],
  })
```